apiserver/common/action.go

```diff
@@ -43,15 +43,16 @@ func TagToActionReceiverFn(findEntity func(names.Tag) (state.Entity, error)) fun
 	return func(tag string) (state.ActionReceiver, error) {
 		receiverTag, err := names.ParseTag(tag)
 		if err != nil {
-			return nil, ErrBadId
+			return nil, errors.NotFoundf("%s", tag)
```

Comment:
this isn't a not found - the tag is invalid
you need to return errors.NotValidf()